pkg/schema/schema.go

```diff
@@ -87,13 +89,13 @@ func (r *Resources) appendResource(resource *Resource) {
 
 // Append appends a single resource to the resource list
 func (r *Resources) Append(resource *Resource) {
-	r.appendResource(resource)
+	r.appendResource(resource, appendUniqueMap)
 }
 
 // Merge merges a list of resources into the main list
 func (r *Resources) Merge(resources *Resources) {
 	for _, item := range resources.Items {
-		r.appendResource(item)
+		r.appendResource(item, mergeUniqueMap)
```

Comment:
Instead of having a global mergeUniqueMap, shouldn't a better idea be to create a new sync.Map on each call to Merge?

Reply:
yes, that is also possible (and better), my thought process the current approach was that we can take advantage of the global map to remove duplicates across multiple resource lists